test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1

```diff
@@ -88,3 +88,25 @@ Describe "Import-Module for Binary Modules in GAC" -Tags 'CI' {
         }
     }
 }
+
+Describe "Import-Module for Binary Modules" -Tags 'CI' {
+
+    It "PS should try to load the assembly from file path first" {
+ $src = @"
+using System;
+namespace ModuleCmdlets
+{ 
+  public class TestBinaryModuleCmdlet1Command
+  {
+  }
+}
+"@
+
+    Add-Type -TypeDefinition $src -OutputAssembly $TESTDRIVE\System.Management.Automation.dll
+    Import-Module $TESTDRIVE\System.Management.Automation.dll -Force
+
+    #Ignore slash format difference under windows/Unix
+    $path = (Get-ChildItem $TESTDRIVE\System.Management.Automation.dll).FullName
+    [System.AppDomain]::CurrentDomain.GetAssemblies().Location.Contains($path) | Should be $true
```

Comment:
Instead of this, you can use `Import-Module -Passthru` and have the exact `Assembly` instance to validate.

We also want to avoid `Should Be $true` because the logs are less useful, so I'd like to see something like:

```powershell
$assem = Import-Module -PassThru ...
$assem.Location | Should Be $Path
```